internal/k8s/controller.go

```diff
@@ -43,6 +43,19 @@ func init() {
 	utilruntime.Must(gw.AddToScheme(scheme))
 }
 
+type ConsulNamespaceConfig struct {
+	ConsulDestinationNamespace      string
+	MirrorKubernetesNamespaces      bool
+	MirrorKubernetesNamespacePrefix string
+}
+
+func (c ConsulNamespaceConfig) Namespace(namespace string) string {
+	if c.MirrorKubernetesNamespaces {
+		return c.MirrorKubernetesNamespacePrefix + namespace
+	}
+	return c.ConsulDestinationNamespace
+}
```

Comment:
So, the main problem I see with build tags, unless they modify some code that's generating namespaces, is that if we say, have some separate distinct tests, we're basically going to want to duplicate all of our testing code in the enterprise tagged build. Personally I'd rather have a few high-level e2e tests that leverage a good amount of our feature set than treat the e2e tests like cheap unit tests.

What I think would be preferable is doing something like generating a namespace of `""` for our OSS Consul deployments, a random namespace for our enterprise deployments, and then adding some code into our e2e tests that takes this namespace into account when verifying the registration on the Consul side of things. This would at least test that our namespace interactions with Consul are working. Our mapping function leveraging prefixes v. static namespaces v. k8s namespaces can just be unit tested from my perspective.